test/heartbeat_tests.py

```diff
@@ -17,13 +14,13 @@ def it_should_send_the_heartbeat(self):
 
     def cleanup_tasks(self):
         self.protocol.heartbeat_monitor.stop()
+        self.loop.run_until_complete(
+            self.protocol.heartbeat_monitor.wait_closed())
```

Comment:
We should wrap this in a timeout. If `wait_closed` hangs then the test will hang.